gradoop-flink/src/main/java/org/gradoop/flink/model/impl/operators/sampling/RandomLimitedDegreeVertexSampling.java

```diff
@@ -0,0 +1,159 @@
+   */
+  private final long degreeThreshold;
+
+  /**
+   * Type of vertex degree to be considered in sampling:
+   * input degree, output degree, sum of both.
+   */
+  private final VertexDegree.DegreeType degreeType;
+
+  /**
+   * Creates new RandomLimitedDegreeVertexSampling instance.
+   *
+   * @param sampleSize relative sample size, e.g. 0.5
+   */
+  public RandomLimitedDegreeVertexSampling(float sampleSize) {
+    this(sampleSize, 0L);
+  }
+
+  /**
+   * Creates new RandomLimitedDegreeVertexSampling instance.
+   *
+   * @param sampleSize relative sample size, e.g. 0.5
+   * @param randomSeed random seed value (can be {@code null})
+   */
+  public RandomLimitedDegreeVertexSampling(float sampleSize, long randomSeed) {
+    this.sampleSize = sampleSize;
+    this.randomSeed = randomSeed;
+    this.degreeThreshold = 2L;
+    this.degreeType = VertexDegree.DegreeType.Degree;
+  }
+
+  /**
+   * Creates new RandomLimitedDegreeVertexSampling instance.
+   *
+   * @param sampleSize relative sample size, e.g. 0.5
+   * @param randomSeed random seed value (can be {@code null}
+   * @param degreeThreshold threshold for degrees of sampled vertices, e.g. 3
+   * @param degreeType type of degree for sampling, e.g. VertexDegree.DegreeType.InputDegree
+   */
+  public RandomLimitedDegreeVertexSampling(float sampleSize, long randomSeed, long degreeThreshold,
+    VertexDegree.DegreeType degreeType) {
+    this.sampleSize = sampleSize;
+    this.randomSeed = randomSeed;
+    this.degreeThreshold = degreeThreshold;
+    this.degreeType = degreeType;
+  }
+
+  /**
+   * Creates new RandomLimitedDegreeVertexSampling instance.
+   *
+   * @param sampleSize relative sample size, e.g. 0.5
+   * @param degreeThreshold threshold for degrees of sampled vertices, e.g. 3
+   * @param degreeType type of degree for sampling, e.g. VertexDegree.DegreeType.InputDegree
+   */
+  public RandomLimitedDegreeVertexSampling(float sampleSize, long degreeThreshold,
+    VertexDegree.DegreeType degreeType) {
+    this.sampleSize = sampleSize;
+    this.randomSeed = 0L;
+    this.degreeThreshold = degreeThreshold;
+    this.degreeType = degreeType;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public LogicalGraph execute(LogicalGraph graph) {
+
+    graph = new DistinctVertexDegrees(VertexDegree.DEGREE_PROPERTY_NAME,
+      VertexDegree.IN_DEGREE_PROPERTY_NAME, VertexDegree.OUT_DEGREE_PROPERTY_NAME,
+      true).execute(graph);
+
+    List<String> unnecessaryPropertyNames = new ArrayList<>();
+    unnecessaryPropertyNames.add(VertexDegree.DEGREE_PROPERTY_NAME);
+    unnecessaryPropertyNames.add(VertexDegree.IN_DEGREE_PROPERTY_NAME);
+    unnecessaryPropertyNames.add(VertexDegree.OUT_DEGREE_PROPERTY_NAME);
+
+    DataSet<Vertex> newVertices = graph.getVertices()
+      .filter(new LimitedDegreeVertexRandomFilter<>(sampleSize, randomSeed, degreeThreshold,
+        degreeType))
+      .map(new RemoveUnnecessaryPropertiesMap<>(unnecessaryPropertyNames));
```

Comment:
Please use the [PropertyRemover](https://github.com/dbs-leipzig/gradoop/blob/master/gradoop-flink/src/main/java/org/gradoop/flink/model/impl/functions/epgm/PropertyRemover.java) instead of `RemoveUnnecessaryPropertiesMap`.  Due to chaining it will be very similar in efficiency.